### ISSUE TITLE
Conditional Access: Error if users condition contains roles "All"

### ISSUE THREAD

**Issue by spa5603** [hashicorp/terraform-provider-azuread#686]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version

terraform -v
Terraform v1.0.9
on linux_amd64
+ provider registry.terraform.io/hashicorp/azuread v2.11.0
+ provider registry.terraform.io/hashicorp/azurerm v2.49.0

### Affected Resource(s)

* `azuread_conditional_access_policy`

### Terraform Configuration Files

```hcl
### Definition of Terraform Version
terraform {
  required_version = "~> 1.0.0"

  ### Definition of required providers
  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 2.49.0"
    }
    azuread = {
      source  = "hashicorp/azuread"
      version = "~> 2.11"
    }

  }
}

### Definition of Terraform provider connections
provider "azurerm" {
  features {
    key_vault {
      purge_soft_delete_on_destroy = true
    }
  }
  tenant_id       = var.tenant_id
  subscription_id = var.subscription_id
}
provider "azuread" {
  tenant_id = var.tenant_id
}

### Conditional Access Policy Definition
resource "azuread_conditional_access_policy" "mfa01" {
  display_name = "some-name"
  state        = "disabled"

  conditions {
    client_app_types = ["browser", "mobileAppsAndDesktopClients"]
    #sign_in_risk_levels = ["none"]
    #user_risk_levels    = ["none"]

    applications {
      included_applications = ["All"]
      #excluded_applications = ["00000004-0000-0ff1-ce00-000000000000"]
    }

    locations {
      included_locations = ["All"]
      #excluded_locations = ["AllTrusted"]
    }

    platforms {
      included_platforms = ["all"]
      #excluded_platforms = ["iOS"]
    }

    users {
      included_roles = ["All"]
    }
  }

  grant_controls {
    operator          = "OR"
    built_in_controls = ["mfa"]
  }
}
```

### Expected Behavior

Conditional Access Policy will be successfully created.
 
### Actual Behavior

`terraform apply` ends with the following error after approve the creation process. This policy is basically the almost the example from the tf registry...

Error: Could not create conditional access policy
│
│   with azuread_conditional_access_policy.mfa01,
│   on conditionalaccess.tf line 1, in resource "azuread_conditional_access_policy" "mfa01":
│    1: resource "azuread_conditional_access_policy" "mfa01" {
│
│ ConditionalAccessPoliciesClient.BaseClient.Post(): unexpected status 400 with OData error: BadRequest: 1102: 'users'
│ condition contains invalid role ids: All.

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by manicminer** [comment]
Hi @spa5603, thanks for reporting this. We recently fixed a couple of casing errors in this resource and I suspect the docs may be out of date. Can you try again using `all` as the value (all lowercase)?

**Comment by spa5603** [comment]
Hi @manicminer thanks for your fast reply on this issue. Unfortunately is the result the same:
 
 Error: Could not create conditional access policy
│
│   with azuread_conditional_access_policy.mfa01,
│   on conditionalaccess.tf line 1, in resource "azuread_conditional_access_policy" "mfa01":
│    1: resource "azuread_conditional_access_policy" "mfa01" {
│
│ ConditionalAccessPoliciesClient.BaseClient.Post(): unexpected status 400 with OData error: BadRequest: 1102: 'users'
│ condition contains invalid role ids: **all.**

**Comment by manicminer** [comment]
Hi @spa5603, thanks for reporting back. Taking a closer look at this, I don't believe this is affected by any of our recent changes. We sourced these values from the [API documentation](https://docs.microsoft.com/en-us/graph/api/resources/conditionalaccessusers?view=graph-rest-beta) which states that the keyword "All" is valid for both `includeGroups` (TF property `included_groups`) and `includeRoles` (TF property `included_roles`).

However I just did some more testing and neither of these accepted those values. I looked through the portal UI too and could not find any option to apply a CA policy to "all roles" or "all groups"; in fact the portal forces you to select at least one of each. So it would appear that this is an API documentation bug that we inadvertently inherited.

![Screenshot 2021-12-02 at 22 30 02](https://user-images.githubusercontent.com/251987/144513649-141aa899-fa54-418f-a0c9-b4cd689ee34e.png)

Unfortunately it doesn't seem like there's anything we can do here, except to update our docs to reflect this.
